#50 Fixed Avoid 2.0 setup wizard but provide secure-by-default configuration

loicortola opened the pull request:
Simply implemented the fixes discussed in issue #50 .
Allows a simple admin account setup.
Tested on debian / ubuntu.

noidi commented on templates/basic-security.groovy:
```diff
@@ -0,0 +1,16 @@
+
+def hudsonRealm = new HudsonPrivateSecurityRealm(false)
+hudsonRealm.createAccount('{{ jenkins_admin.username }}','{{ jenkins_admin.password }}')
+instance.setSecurityRealm(hudsonRealm)
```
Aren't the scripts under `/var/lib/jenkins/init.groovy.d/` run on *every* startup and not just the first? If so, I think this script will remove any users that may have been added since the last role application. Also, if the installation is configured to use [matrix-based security](https://wiki.jenkins-ci.org/display/JENKINS/Matrix-based+security), that configuration will be lost when the authorization strategy is replaced.

loicortola commented:
Hi @noidi , mmmh good point.
I will make the changes in the weekend.

I also agree that using matrix-based setups will override this strategy, 
perhaps should I allow an extra option which enables both the matrix-feature and adds the admin as a default entry. For basic use, I think the default should remain as is though.

What do you think?

loicortola commented:
Updated the groovy script to be processed only when security is disabled, and added task to automatically remove script when Jenkins is ready and plugins installed.

@noidi this solves the original problem. Switching to matrix-based security wont be overridden when Jenkins is restarted.

noidi commented:
I haven't given this a try yet, but looking at the commit the new version seems to fix the issues I reported. :+1:

noidi commented on tasks/main.yml:
```diff
@@ -46,3 +46,7 @@
+
+# Remove init.groovy.d/basic-security settings
+- name: Remove Jenkins security init scripts
+  shell: "rm -f /var/lib/jenkins/init.groovy.d/basic-security.groovy"
```
I think this task is causing the idempotence test failure on Travis. It should probably be changed to remove the file using the [file module](http://docs.ansible.com/ansible/file_module.html).

loicortola replied:
Will do.

loicortola commented:
Hi all,

I have successfully tested the PR and it works correctly.
However, as the user creation process is not idempotent anymore (I guess the process must output a random salt somewhere), the travis test fails.

As I am really not at ease with this tool, any help or advice is welcome. Thoughts?

fiunchinho commented:
Any news?

loicortola commented:
I propose two solutions:
1) Someone knows how to fix the idempotence test
2) Remove the idempotence test (not my favourite solution)

Anyhow, the request should be mergeable as-is (technically speaking, it works)

geerlingguy commented:
I'm going to take a look and see whether this or #56 is the right approach, and try to get it working...

loicortola commented:
Thanks for merging and performing latest cleaning in #65

geerlingguy commented:
No problem, thanks for opening this PR!
